src/components/postgresConnect.ts

```diff
@@ -659,6 +659,116 @@ class PostgresDB {
       throw err
     }
   }
+
+  public getNumReviewLevels = async (templateId: number, stageNumber: number) => {
+    const text = `
+    SELECT MAX(level) FROM template_permission
+    WHERE template_id = $1 
+    AND stage_number = $2
+    `
+    try {
+      const result = await this.query({ text, values: [templateId, stageNumber] })
+      return result.rows[0].max
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  }
+
+  public getCurrentReviewLevel = async (reviewId: number) => {
+    const text = `SELECT level FROM review WHERE id = $1`
+    try {
+      const result = await this.query({ text, values: [reviewId] })
+      return result.rows[0].level
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  }
+
+  public getReviewersForApplicationStageLevel = async (
+    templateId: number,
+    stageNumber: number,
+    reviewLevel: number
+  ) => {
+    const text = `
+    SELECT user_id, organisation_id, restrictions FROM 
+    permission_join pj JOIN template_permission tp
+    ON pj.permission_name_id = tp.permission_name_id
+    WHERE template_id = $1
+    AND stage_number = $2
+    AND level = $3
+    `
+    try {
+      const result = await this.query({
+        text,
+        values: [templateId, stageNumber, reviewLevel],
+      })
+      return result.rows
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  }
+
+  public addReviewAssignments = async (reviewAssignments: any) => {
+    const reviewAssignmentIds = []
+    for (const reviewAssignment of reviewAssignments) {
+      const {
+        reviewerId,
+        orgId,
+        stageId,
+        stageNumber,
+        status,
+        applicationId,
+        templateSectionRestrictions,
+        level,
+        isLastLevel,
+      } = reviewAssignment
+      // Needs a slightly different query with different CONFLICT restrictions
+      // depending on whether orgId exists or not.
+      const text = `
+        INSERT INTO review_assignment (
+          reviewer_id,${orgId ? ' organisation_id,' : ''}
+          stage_id, stage_number, status, application_id,
+          template_section_restrictions, level, is_last_level
+          )
+        VALUES ($1, $2, $3, $4, $5, $6, $7, $8${orgId ? ', $9' : ''})
+        ON CONFLICT (reviewer_id,${
```

Comment:
great this would prevent already 'assigned' review_assignments from changing status. Would be good to add a quick comment about this above